app/src/main/java/com/android/unio/model/event/EventViewModel.kt

```diff
@@ -317,21 +326,86 @@ constructor(
           eventUserPictureRepository.addEventUserPicture(
               newEventPicture,
               {
-                event.eventPictures.add(newEventPicture.uid)
+                event.eventPictures.add(newEventPicture)
                 updateEventWithoutImage(
                     event,
-                    { event.eventPictures.add(newEventPicture) },
+                    { onSuccess() },
                     { e ->
                       Log.e("EventViewModel", "An error occurred while updating an event: $e")
-                    })
+                    },
+                    false)
               },
               { e ->
+                onFailure(e)
                 Log.e("EventViewModel", "An error occurred while adding an event picture: $e")
               })
         },
         onFailure = { e -> Log.e("ImageRepository", "Failed to store image: $e") })
   }
 
+  /**
+   * Update an existing eventUserPicture without updating its image.
+   *
+   * @param event The event in question.
+   * @param picture The [EventUserPicture] to update
+   */
+  fun updateEventUserPictureWithoutImage(
+      event: Event,
+      picture: EventUserPicture,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    eventUserPictureRepository.addEventUserPicture(
+        picture,
+        {
+          if (!event.eventPictures.contains(picture.uid)) {
+            event.eventPictures.add(picture)
+          }
+          updateEventWithoutImage(
+              event,
+              {
+                _events.value = _events.value.map { if (it.uid == event.uid) event else it }
+                onSuccess()
+              },
+              { e ->
+                onFailure(e)
+                Log.e("EventViewModel", "An error occurred while updating an event: $e")
+              },
+              false)
+        },
+        { e ->
+          onFailure(e)
+          Log.e("EventViewModel", "An error occurred while adding an event picture: $e")
+        })
+  }
+
+  fun deleteEventUserPicture(
+      uid: String,
+      event: Event,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+
+    eventUserPictureRepository.deleteEventUserPictureById(
+        uid,
+        {
+          event.eventPictures.remove(uid)
+          updateEventWithoutImage(
```

Comment:
Instead of doing this - that will in firebase for every field to change, you could have just override the image field to "" to avoid unnecessary server computing time